airflow/contrib/hooks/winrm_hook.py

```diff
@@ -31,20 +31,53 @@ class WinRMHook(BaseHook, LoggingMixin):
-        for parameter values.
+    :seealso: https://github.com/diyan/pywinrm/blob/master/winrm/protocol.py
 
-    :param str ssh_conn_id: connection id from airflow Connections from where all
+    :param ssh_conn_id: connection id from airflow Connections from where all
         the required parameters can be fetched like username and password.
         Thought the priority is given to the param passed during init
-    :param str remote_host: Remote host to connect to.
+    :type ssh_conn_id: string
+    :param endpoint: When set to `None`, endpoint will be constructed like this:
+        'http://{remote_host}:{remote_port}/wsman'
+    :type endpoint: string
+    :param remote_host: Remote host to connect to.
         Ignored if `endpoint` is not `None`.
-    :param int remote_port: Remote port to connect to.
+    :type remote_host: string
+    :param remote_port: Remote port to connect to.
         Ignored if `endpoint` is not `None`.
-    :param str endpoint: When set to `None`, endpoint will be constructed like this:
-        'http://{remote_host}:{remote_port}/wsman'
-    :param str username: username to connect to the remote_host
-    :param str password: password of the username to connect to the remote_host
+    :type remote_port: int
+    :param transport: transport type, one of 'plaintext' (default), 'kerberos', 'ssl', 'ntlm', 'credssp'
+    :type transport: string
+    :param username: username to connect to the remote_host
+    :type username: string
+    :param password: password of the username to connect to the remote_host
+    :type password: string
+    :param service: the service name, default is HTTP
+    :type service: string
+    :param keytab: the path to a keytab file if you are using one
+    :type keytab: string
+    :param ca_trust_path: Certification Authority trust path
+    :type ca_trust_path: string
+    :param cert_pem: client authentication certificate file path in PEM format
+    :type cert_pem: string
+    :param cert_key_pem: client authentication certificate key file path in PEM format
+    :type cert_key_pem: string
+    :param server_cert_validation: whether server certificate should be validated on Python versions that suppport it; one of 'validate' (default), 'ignore' #NOQA
+    :type server_cert_validation: string
+    :param kerberos_delegation: if True, TGT is sent to target server to allow multiple hops
+    :type kerberos_delegation: bool
+    :param read_timeout_sec: maximum seconds to wait before an HTTP connect/read times out (default 30). This value should be slightly higher than operation_timeout_sec, as the server can block *at least* that long.
+    :type read_timeout_sec: int
+    :param operation_timeout_sec: maximum allowed time in seconds for any single wsman HTTP operation (default 20). Note that operation timeouts while receiving output (the only wsman operation that should take any significant time, and where these timeouts are expected) will be silently retried indefinitely.
```

Comment:
You will need to break such big lines into multiple lines such as below:

https://github.com/apache/incubator-airflow/blob/ef3ead167be9d7dd72f75da9d1f4e00438b8ea6e/airflow/contrib/operators/bigquery_get_data.py#L65-L67

Otherwise this would fail the flake8 test.

Reply:
Fixed